restorer.go

```diff
@@ -1,34 +1,50 @@
 package lifecycle
 
 import (
+	"fmt"
+	"path/filepath"
+
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 
+	"github.com/buildpacks/imgutil"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/launch"
 	"github.com/buildpacks/lifecycle/layers"
 )
 
 type Restorer struct {
-	LayersDir  string
-	Buildpacks []GroupBuildpack
-	Logger     Logger
+	LayersDir   string
+	Buildpacks  []GroupBuildpack
+	Logger      Logger
+	SkipLayers  bool
+	PlatformAPI *api.Version
 }
 
 // Restore attempts to restore layer data for cache=true layers, removing the layer when unsuccessful.
 // If a usable cache is not provided, Restore will remove all cache=true layer metadata.
-func (r *Restorer) Restore(cache Cache) error {
+func (r *Restorer) Restore(img imgutil.Image, cache Cache) error {
```

Comment:
Where does the image come from for earlier platform APIs?

Reply:
it would be `nil` and unused in the earlier platform API code path. I don't like it and would love to get feedback on a better path forward. You can see in `creator` a similar ugly pattern emerging with `analyze(lifecycle.BuildpackGroup{}, nil)` in one code path of the `analyzer` execution.

Reply:
I do wonder if there's a way this could be architected differently to make it a bit cleaner. Thinking...

Reply:
Would there be value in creating a `lifecycle.LayerAnalyzer`?